libcudacxx/include/cuda/__tma/make_tma_descriptor.h

```diff
@@ -0,0 +1,575 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the libcu++ Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _CUDA___TMA_MAKE_TMA_DESCRIPTOR_H
+#define _CUDA___TMA_MAKE_TMA_DESCRIPTOR_H
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#if _CCCL_HAS_INCLUDE(<dlpack/dlpack.h>)
+
+#  include <cuda/__driver/driver_api.h>
+#  include <cuda/__memory/is_aligned.h>
+#  include <cuda/std/__algorithm/min.h>
+#  include <cuda/std/__memory/is_sufficiently_aligned.h>
+#  include <cuda/std/array>
+#  include <cuda/std/cstddef>
+#  include <cuda/std/cstdint>
+#  include <cuda/std/limits>
+#  include <cuda/std/span>
+
+#  include <dlpack/dlpack.h>
+//
+#  include <cuda/std/__cccl/prologue.h>
+
+_CCCL_BEGIN_NAMESPACE_CUDA
+
+static_assert(DLPACK_MAJOR_VERSION == 1, "DLPACK_MAJOR_VERSION must be 1");
+
+/***********************************************************************************************************************
+ * Public Enums
+ ***********************************************************************************************************************/
+
+enum class tma_oob_fill
+{
+  none,
+  nan
+};
```

Comment:
Q: Should we assign the enumerators to the same values as the driver macros?
```suggestion
enum class tma_oob_fill
{
  none = ::CU_TENSOR_MAP_FLOAT_OOB_FILL_NONE,
  nan = ::CU_TENSOR_MAP_FLOAT_OOB_FILL_NAN_REQUEST_ZERO_FMA
};
```
?

Reply:
that's a good point. Hope I'm not missing any implication about this dependency